drivers/memc/memc_mcux_xspi.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	nxp_xspi
+
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <soc.h>
+#include "memc_mcux_xspi.h"
+
+LOG_MODULE_REGISTER(memc_mcux_xspi, CONFIG_MEMC_LOG_LEVEL);
+
+struct memc_mcux_xspi_config {
+	const struct pinctrl_dev_config *pincfg;
+	xspi_config_t xspi_config;
+};
+
+struct memc_mcux_xspi_data {
+	XSPI_Type *base;
+	bool xip;
+	uint32_t amba_address;
+	const struct device *clock_dev;
+	clock_control_subsys_t clock_subsys;
+};
+
+int memc_mcux_xspi_get_root_clock(const struct device *dev, uint32_t *clock_rate)
+{
+	struct memc_mcux_xspi_data *data = dev->data;
+
+	return clock_control_get_rate(data->clock_dev, data->clock_subsys, clock_rate);
+}
+
+void memc_xspi_wait_bus_idle(const struct device *dev)
+{
+	struct memc_mcux_xspi_data *data = dev->data;
+
+	while (!XSPI_GetBusIdleStatus(data->base)) {
+	}
+}
+
+int memc_xspi_set_device_config(const struct device *dev,
+		const xspi_device_config_t *device_config,
+		const uint32_t *lut_array,
+		uint8_t lut_count)
+{
+	struct memc_mcux_xspi_data *data = dev->data;
+	XSPI_Type *base = data->base;
+	status_t status;
+
+	/* Configure flash settings according to serial flash feature. */
+	status = XSPI_SetDeviceConfig(base, (xspi_device_config_t *)device_config);
+	if (status != kStatus_Success) {
+		LOG_ERR("XSPI_SetDeviceConfig failed with status %u\n", status);
+		return -ENODEV;
+	}
+
+	XSPI_UpdateLUT(base, 0, lut_array, lut_count);
+
+	return 0;
+}
+
+uint32_t memc_mcux_xspi_get_ahb_address(const struct device *dev)
+{
+	return ((const struct memc_mcux_xspi_data *)dev->data)->amba_address;
+}
+
+int memc_mcux_xspi_transfer(const struct device *dev, xspi_transfer_t * xfer)
+{
+	XSPI_Type *base = ((struct memc_mcux_xspi_data *)dev->data)->base;
+	status_t status;
+
+	status = XSPI_TransferBlocking(base, xfer);
+
+	return (status == kStatus_Success) ? 0 : -EIO;
+}
+
+bool memc_xspi_is_running_xip(const struct device *dev)
+{
+	return ((const struct memc_mcux_xspi_data *)dev->data)->xip;
+}
+
+static int memc_mcux_xspi_init(const struct device *dev)
+{
+	const struct memc_mcux_xspi_config * memc_xspi_config = dev->config;
+	const struct pinctrl_dev_config *pincfg = memc_xspi_config->pincfg;
+	xspi_config_t config = memc_xspi_config->xspi_config;
+	XSPI_Type *base = ((struct memc_mcux_xspi_data *)dev->data)->base;
+	int ret;
+
+	if (memc_xspi_is_running_xip(dev)) {
+		if (!IS_ENABLED(CONFIG_MEMC_MCUX_XSPI_INIT_XIP)) {
+			LOG_ERR("XIP active on %s, skipping init", dev->name);
+			return 0;
+		}
+	}
+
+	ret = pinctrl_apply_state(pincfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (config.ptrAhbAccessConfig != NULL) {
```

Comment:
No other configurations to set ptrAhbAccessConfig as NULL, this if condition can be removed